Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2772,6 +2772,44 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+
+bool CSSPropertyParser::consumeAnimationRangeShorthand(bool important)
+{
+    // FIXME: Add extra handling for animation range sequences.
+    RefPtr start = consumeAnimationRangeStart(m_range, m_context);
+    if (!start)
+        return false;
+
+    RefPtr<CSSValue> end;
+    m_range.consumeWhitespace();
+    if (m_range.atEnd()) {
+        // From the spec: If <'animation-range-end'> is omitted and <'animation-range-start'> includes a component, then
+        // animation-range-end is set to that same and 100%. Otherwise, any omitted longhand is set to its initial value.
+
+        auto rangeEndValueForStartValue = [](const CSSValue& value) {
+            if (is<CSSPrimitiveValue>(value) && SingleTimelineRange::isOffsetValue(downcast<CSSPrimitiveValue>(value)))
```

Comment:
The common pattern is this:

```suggestion
            RefPtr primitiveValue = dynamicDowncast<CSSPrimitiveValue>(value);
            if (primitiveValue && SingleTimelineRange::isOffsetValue(primitiveValue))
```

Reply:
Fixed.